hazelcast/src/main/java/com/hazelcast/config/YamlConfigBuilder.java

```diff
@@ -144,7 +165,139 @@ private void parseAndBuildConfig(Config config) throws Exception {
+            if (targetChild != null) {
+                merge(sourceChild, targetChild);
+            } else {
+                ((MutableYamlMapping) targetAsMapping).addChild(sourceChild.nodeName(), sourceChild);
+            }
+        }
+    }
+
+    private void replaceVariables(Node node) throws Exception {
+        // if no config-replacer is defined, use backward compatible default behavior for missing properties
+        boolean failFast = false;
+
+        List<ConfigReplacer> replacers = new ArrayList<ConfigReplacer>();
+
+        // Always use the Property replacer first.
+        PropertyReplacer propertyReplacer = new PropertyReplacer();
+        propertyReplacer.init(properties);
+        replacers.add(propertyReplacer);
+
+        // Add other replacers
+        Node replacersNode = node.getAttributes().getNamedItem(ConfigSections.CONFIG_REPLACERS.name);
+
+        if (replacersNode != null) {
+            String failFastAttr = getAttribute(replacersNode, "fail-if-value-missing", true);
+            failFast = isNullOrEmpty(failFastAttr) || Boolean.parseBoolean(failFastAttr);
+            for (Node n : childElements(replacersNode)) {
+                String nodeName = cleanNodeName(n);
+                if ("replacers".equals(nodeName)) {
+                    for (Node replacerNode : childElements(n)) {
+                        replacers.add(createReplacer(replacerNode));
+                    }
+                }
+            }
+        }
+
+        ConfigReplacerHelper.traverseChildrenAndReplaceVariables(node, replacers, failFast);
+    }
+
+    private ConfigReplacer createReplacer(Node node) throws Exception {
+        String replacerClass = getAttribute(node, "class-name", true);
+        Properties properties = new Properties();
+        for (Node n : childElements(node)) {
+            String value = cleanNodeName(n);
+            if ("properties".equals(value)) {
+                fillReplacerProperties(n, properties);
+            }
+        }
+        ConfigReplacer replacer = (ConfigReplacer) Class.forName(replacerClass).newInstance();
+        replacer.init(properties);
+        return replacer;
+    }
+
+    public YamlConfigBuilder setProperties(Properties properties) {
+        this.properties = properties;
+        return this;
+    }
+
+    private void fillReplacerProperties(Node node, Properties properties) {
+        YamlMapping propertiesMapping = asMapping(((ElementAdapter) node).getYamlNode());
+        for (YamlNameNodePair childNodePair : propertiesMapping.childrenPairs()) {
+            String childName = childNodePair.nodeName();
+            YamlNode child = childNodePair.childNode();
+            if (child != null) {
+                Object nodeValue = asScalar(child).nodeValue();
+                properties.put(childName, nodeValue != null ? nodeValue.toString() : "");
+            } else {
+                properties.put(childName, "");
```

Comment:
Does any of your tests cover this line?

Reply:
It was, but I changed the test make the flow coming here, so removed the `if` and the complete `else` block. See my comment here: https://github.com/hazelcast/hazelcast/pull/14451#discussion_r257462487